apache/metron
# METRON-1725: Add ability to specify YARN queue for pcap jobs

## Contributor Comments
This PR exposes a configuration option in the Pcap CLI and REST app for the YARN queue a pcap job will be submitted to.  This configuration is passed on to the `mapreduce.job.queuename` property in the Hadoop config.  A YARN queue is set in the Pcap CLI with an additional command line option (`-yq`) and is set in the REST app as a spring property (`pcap.yarn.queue`).

### Testing
This has been tested in full dev:

1. Create a YARN queue named `pcap` using the YARN Queue Manager
2. Ambari should prompt you, but be sure to restart the Resource Manager
3. Submit a job with the Pcap CLI and add the YARN queue option:  `-yq pcap`
4. While the job is running navigate to the Resource Manager UI and verify the job is running in the `pcap` queue
5. Try to submit a job to a queue that doesn't exist.  You should get a clear error stating the queue is missing
6. Go to Ambari > Services > Metron > REST and set the `Metron Spring options` property to `--pcap.yarn.queue=pcap`. 
7. Ambari will prompt you to restart REST
8. Submit a job through the REST app and verify the job is running in the `pcap` queue
9. Submitting jobs in the Pcap CLI without the `yq` option and submitting jobs in REST without setting the `pcap.yarn.queue` should submit to the default queue.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
